File: backend-java/test-design-assistant-core/src/main/java/com/sinosoft/testdesign/entity/TestReport.java
package com.sinosoft.testdesign.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.time.LocalDateTime;

/**
 * 测试报告实体
 * 
 * @author sinosoft
 * @date 2024-01-17
 */
@Data
@Entity
@Table(name = "test_report")
public class TestReport {
    
    /**
     * 主键ID
     */
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    /**
     * 报告编码（RPT-YYYYMMDD-序号）
     */
    @Column(name = "report_code", unique = true, nullable = false, length = 100)
    private String reportCode;
    
    /**
     * 报告名称
     */
    @Column(name = "report_name", nullable = false, length = 500)
    private String reportName;
    
    /**
     * 报告类型：EXECUTION/COVERAGE/QUALITY/RISK
     */
    @Column(name = "report_type", nullable = false, length = 50)
    private String reportType;
    
    /**
     * 报告模板ID
     */
    @Column(name = "template_id")
    private Long templateId;
    
    /**
     * 关联需求ID
     */
    @Column(name = "requirement_id")
    private Long requirementId;
    
    /**
     * 关联执行任务ID
     */
    @Column(name = "execution_task_id")
    private Long executionTaskId;
    
    /**
     * 报告内容（JSON格式）
     */
    @Column(name = "report_content", columnDefinition = "TEXT")
    private String reportContent;
    
    /**
     * 报告摘要
     */
    @Column(name = "report_summary", columnDefinition = "TEXT")
    private String reportSummary;
    
    /**
     * 报告状态：DRAFT/PUBLISHED
     */
    @Column(name = "report_status", nullable = false, length = 50)
    private String reportStatus = "DRAFT";
    
    /**
     * 生成配置（JSON格式）
     */
    @Column(name = "generate_config", columnDefinition = "TEXT")
    private String generateConfig;
    
    /**
     * 报告文件URL（Word/PDF）
     */
    @Column(name = "file_url", length = 1000)
    private String fileUrl;
    
    /**
     * 文件格式：WORD/PDF/EXCEL
     */
    @Column(name = "file_format", length = 50)
    private String fileFormat;
    
    /**
     * 创建人ID
     */
    @Column(name = "creator_id")
    private Long creatorId;
    
    /**
     * 创建人姓名
     */
    @Column(name = "creator_name", length = 100)
    private String creatorName;
    
    /**
     * 创建时间
     */
    @Column(name = "create_time")
    private LocalDateTime createTime;
    
    /**
     * 更新时间
     */
    @Column(name = "update_time")
    private LocalDateTime updateTime;
    
    /**
     * 发布时间
     */
    @Column(name = "publish_time")
    private LocalDateTime publishTime;
    
    @PrePersist
    public void prePersist() {
        this.createTime = LocalDateTime.now();
        this.updateTime = LocalDateTime.now();
    }
    
    @PreUpdate
    public void preUpdate() {
        this.updateTime = LocalDateTime.now();
    }
}

File: frontend/src/api/testReport.ts
import request from './request'

export interface TestReportRequestDTO {
    reportName: string
    reportType: 'EXECUTION' | 'COVERAGE' | 'QUALITY' | 'RISK'
    templateId?: number
    requirementId?: number
    executionTaskId?: number
    generateConfig?: string
    creatorId?: number
    creatorName?: string
}

export interface TestReportResponseDTO {
    id: number
    reportCode: string
    reportName: string
    reportType: string
    templateId?: number
    requirementId?: number
    executionTaskId?: number
    reportContent?: string
    reportSummary?: string
    reportStatus: 'DRAFT' | 'PUBLISHED'
    generateConfig?: string
    fileUrl?: string
    fileFormat?: string
    creatorId?: number
    creatorName?: string
    createTime?: string
    updateTime?: string
    publishTime?: string
}

export interface PageResult<T> {
    content: T[]
    totalElements: number
    totalPages: number
    size: number
    number: number
}

export const testReportApi = {
    // Generate report
    generateReport(data: TestReportRequestDTO) {
        return request.post<any, TestReportResponseDTO>('/v1/test-reports', data)
    },

    // Get report list
    getReportList(page: number = 0, size: number = 10) {
        return request.get<any, PageResult<TestReportResponseDTO>>('/v1/test-reports', {
            params: { page, size }
        })
    },

    // Get report detail
    getReportById(id: number) {
        return request.get<any, TestReportResponseDTO>(`/v1/test-reports/${id}`)
    },

    // Get report by code
    getReportByCode(reportCode: string) {
        return request.get<any, TestReportResponseDTO>(`/v1/test-reports/code/${reportCode}`)
    },

    // Get reports by requirement ID
    getReportsByRequirementId(requirementId: number) {
        return request.get<any, TestReportResponseDTO[]>(`/v1/test-reports/requirement/${requirementId}`)
    },

    // Get reports by execution task ID
    getReportsByExecutionTaskId(executionTaskId: number) {
        return request.get<any, TestReportResponseDTO[]>(`/v1/test-reports/execution-task/${executionTaskId}`)
    },

    // Update report
    updateReport(id: number, data: TestReportRequestDTO) {
        return request.put<any, TestReportResponseDTO>(`/v1/test-reports/${id}`, data)
    },

    // Publish report
    publishReport(id: number) {
        return request.put<any, TestReportResponseDTO>(`/v1/test-reports/${id}/publish`)
    },

    // Delete report
    deleteReport(id: number) {
        return request.delete<any, void>(`/v1/test-reports/${id}`)
    },

    // Export report
    exportReport(reportCode: string, format: 'WORD' | 'PDF' | 'EXCEL') {
        return request.get<any, string>(`/v1/test-reports/${reportCode}/export`, {
            params: { format }
        })
    },

    // Summarize execution results
    summarizeExecutionResults(requirementId?: number, executionTaskId?: number) {
        return request.get<any, string>('/v1/test-reports/summarize', {
            params: { requirementId, executionTaskId }
        })
    }
}
